.github/workflows/deploy.yml

```diff
@@ -0,0 +1,262 @@
+              "keda-manager") VAR_BRANCH="${{ vars.KEDA_MANAGER_BRANCH }}" ;;
+              "cloud-manager") VAR_BRANCH="${{ vars.CLOUD_MANAGER_BRANCH }}" ;;
+              "docker-registry") VAR_BRANCH="${{ vars.DOCKER_REGISTRY_BRANCH }}" ;;
+              "busola") VAR_BRANCH="${{ vars.BUSOLA_BRANCH }}" ;;
+              "cli") VAR_BRANCH="${{ vars.CLI_BRANCH }}" ;;
+              "registry-proxy") VAR_BRANCH="${{ vars.REGISTRY_PROXY_BRANCH }}" ;;
+              *) VAR_BRANCH="" ;;
+            esac
+            
+            if [ -n "$VAR_BRANCH" ]; then
+              BRANCH="$VAR_BRANCH"
+              echo "✅ Using branch '$BRANCH' from GitHub variable for $REPO"
+            else
+              echo "ℹ️  Using default branch 'main' for $REPO"
+            fi
+          fi
+          
+          echo "branch=$BRANCH" >> $GITHUB_OUTPUT
+          echo "🎯 Final decision: Using branch '$BRANCH' for repository '$REPO'"
+
+      - name: '🗂️ Checkout ${{matrix.repository}} (branch: ${{steps.get-branch.outputs.branch}})'
+        uses: actions/checkout@v4
+        with:
+          repository: kyma-project/${{matrix.repository}}
+          ref: ${{steps.get-branch.outputs.branch}}
+          path: ${{matrix.repository}}
+
+      - name: 🚢 Copy for main repo
+        run: |
+          SOURCE_PATH="${{matrix.repository}}/docs/user/"
+          TARGET_PATH="./docs/external-content/${{matrix.repository}}/docs/"
+          SOURCE_PATH_ASSETS="${{matrix.repository}}/docs/assets/"
+          TARGET_PATH_ASSETS="./docs/external-content/${{matrix.repository}}/docs/"
+          if [ -d "$SOURCE_PATH" ]; then
+            echo "📁 found in ${{matrix.repository}}, copy to $TARGET_PATH"
+            mkdir -p "$TARGET_PATH"
+            cp -rf "$SOURCE_PATH/" "$TARGET_PATH/"
+            if [ -d "$SOURCE_PATH_ASSETS" ]; then
+              echo "↳📁 found in ${{matrix.repository}}, copy to $TARGET_PATH_ASSETS"
+              mkdir -p "$TARGET_PATH_ASSETS"
+              cp -rf "$SOURCE_PATH_ASSETS/" "$TARGET_PATH_ASSETS/"
+            else
+              echo "↳🚫 no folder  docs/assets in ${{matrix.repository}}"
+            fi
+          else
+            echo "🚫 No folder docs/users in ${{matrix.repository}}"
+          fi
+      
+      - name: 📦 Upload copied docs
+        uses: actions/upload-artifact@v4
+        with:
+          name: external-docs-${{matrix.repository}}
+          path: ./docs/external-content/${{matrix.repository}}
+
+  build-and-deploy:
+    name: 🛠️ Build and Deploy VitePress
+    runs-on: ubuntu-latest
+    needs: copy-docs
+    steps:
+      - name: 🗂️ Checkout repository
+        uses: actions/checkout@v4
+        
+      - name: 📥 Download copied docs
+        uses: actions/download-artifact@v4
+        with:
+          name: external-docs-btp-manager
+          path: docs/external-content/btp-manager
+
+      - name: 📥 Download copied docs
```

Comment:
# This single step replaces all 'download-artifact' steps.
      # It downloads all artifacts from the 'copy-docs' job into the specified path.
      - name: 📥 Download all external docs
        uses: actions/download-artifact@v4
        with:
          path: docs/external-content